lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -1711,27 +1711,36 @@ void GrandCentralPass::runOnOperation() {
   instancePaths = &instancePathCache;
 
   /// Contains the set of modules which are instantiated by the DUT, but not a
-  /// companion or instantiated by a companion.  If no DUT exists, treat the top
-  /// module as if it were the DUT.  This works by doing a depth-first walk of
-  /// the instance graph, starting from the "effective" DUT and stopping the
-  /// search at any modules which are known companions.
+  /// companion, instantiated by a companion, or instantiated under a bind.  If
+  /// no DUT exists, treat the top module as if it were the DUT.  This works by
+  /// doing a depth-first walk of the instance graph, starting from the
+  /// "effective" DUT and stopping the search at any modules which are known
+  /// companions or any instances which are marked "lowerToBind".
   DenseSet<igraph::ModuleOpInterface> dutModules;
   FModuleOp effectiveDUT = dut;
   if (!effectiveDUT)
     effectiveDUT = cast<FModuleOp>(
         *instancePaths->instanceGraph.getTopLevelNode()->getModule());
-  auto dfRange =
-      llvm::depth_first(instancePaths->instanceGraph.lookup(effectiveDUT));
-  for (auto i = dfRange.begin(), e = dfRange.end(); i != e;) {
-    auto module = cast<FModuleLike>(*i->getModule());
-    if (AnnotationSet(module).hasAnnotation(companionAnnoClass)) {
-      i.skipChildren();
-      continue;
+  {
+    SmallVector<igraph::ModuleOpInterface> modules({effectiveDUT});
+    while (!modules.empty()) {
+      auto m = modules.pop_back_val();
+      for (InstanceRecord *a : *instancePaths->instanceGraph.lookup(m)) {
```

Comment:
Could the worklist be InstanceRecord*'s instead, to avoid looking them up as this walks?

Reply:
Yes, there's a better organization of this to use `InstanceGraphNode *`. Will switch to that. Nice suggestion.